### DISCUSSION TITLE
A place for help, ideas, etc.

### DISCUSSION THREAD

**Discussion by hrfee** [hrfee/jfa-go#43] [General]
Hi,
Just noticed this feature and thought it'd be a good place for people to put ideas for the project and ask for help with it, rather than creating issues.

**Comment by landcraft** [comment]
I was just about to create an issue when I spotted this! thanks!

I'm running docker on Synology and I'm having trouble with your container.

I get the following error in the logs (over and over)
`standard_init_linux.go:222: exec user process caused "exec format error"
`

Any advice on how to solve? I tried setting up GID & UID as environment variables (I'm using docker-compose) but no luck

**Reply by hrfee** [reply]
This is because your CPU architecture is different from what the executable was compiled for. If you're running on an rpi or some other arm board, I recently set up builds for arm64/armv7, although currently they're only available on the `unstable` docker tag.

**Reply by hrfee** [reply]
Ah also just realised the `latest` tag only has an ARM build. I guess i messed it up when i was experimenting with multiarch builds. Putting an amd64 one up now, so it should be up in ~20m

**Comment by ClankJake** [comment]
You have the possibility to add a timer for the account. Example 30 days of access after that period the account is deactivated. would that possibility?

**Reply by hrfee** [reply]
Sorry I didn't respond to this, I've been working on it for the past couple days and just finished it (i think) in 1ec5d2c. 

There is now a "user expiry" tab on the invite page, click it then press enable and put in the time. When a new user is created with that invite, their account will be disabled/deleted after the specified amount of time. When this happens, an email will be sent to tell them their account has been deactivated. You can change if it disables or deletes the account and whether it should send an email in Settings > User Expiry. 

Also, there's quite a few new translations for this feature, would really appreciate if you helped again.

**Reply by ClankJake** [reply]
You can leave it tomorrow I will translate. I saw now that there was more content for translation.

**Comment by tuhinadri** [comment]


**Reply by hrfee** [reply]
There's a command on the README that you can run:
```bash
docker create
             --name "jfa-go"
             -p 8056:8056
             -v <path to a folder where jfa-go can store its configuration files and data>:/data 
             -v <path to jellyfin config directory>:/jf
             -v /etc/localtime:/etc/localtime:ro
             hrfee/jfa-go
```
Put all of those on the same line, they're only separated for readability.
I assume you don't run Jellyfin in docker yet, so `<path to jellyfin config directory>` should be something like `/etc/jellyfin` (it should contain `system.xml` and some other files). If it isn't, you can try to run a password reset in Jellyfin, and a window should pop up with a path to a file called something like `passwordreset-*.json`. Everything before `passwordreset-*.json` would be the config directory.

Once you've done that, you can run `docker start/stop/restart jfa-go` to start/stop/restart. If you want to update, you need to:
* stop the container (`docker stop jfa-go`)
* remove it with `docker rm jfa-go`
* pull the new container (`docker pull hrfee/jfa-go`)
* re-run the above `docker create` command, then `docker start jfa-go`.

This process is similar with most other containers, such as Jellyfin/Sonarr/Radarr, so it should be quick to pick it up.

**Reply by hrfee** [reply]
> I got an error message cause my jellyfin server is on other hosting & trying to add jfa-go to in another hosting droplet! how I link? & how should I give the username & password of the jellyfin server directory??

<del>`/.app/jellyfin` is the correct path, so in the `docker create` command you would put `-v /.app/jellyfin:/jf`.</del> So your Jellyfin is on a different host to jfa-go? If yes, then you need to set up something for remote access, such as an SMB server. You would then need to mount it to a folder on the droplet, and then give jfa-go the path `/.app/jellyfin` is mounted to.

**Reply by hrfee** [reply]
> Is there anyway I can directly upload the files to my jellyfin server via ssh? Like I downloaded the files in local 1st then change the codes as I need with code editor! then reupload it?
> 
> Now the problem is where should I put the jellyfin path! I just download the codes from githiub as a zip file!
> 
> & after uploading it which path should I visit to get the sign up 
What files are you downloading and editing? If you're going to run jfa-go in docker, you don't need to download anything. When you run the `docker create` command, docker will automatically download the container.

**Reply by hrfee** [reply]
> & I have a request! Suppose I wanted to change my jellyfin address but want all users of my jellyfin server could sign in with the previous info? Can I do it? So
> I have to copy-paste any config file in th new path of jellyfin?

I don't really understand the question. Do you mean that you're changing the address you access Jellyfin from, but you want the old address to redirect to the new one?

**Reply by hrfee** [reply]
This doesn't really work like a wordpress plugin. It's written in Go, which is a compiled language, meaning you can't just edit the code then put it on your server and run it. If you really want to modify the code, then you can download the source code, and compile it by referring to the instructions [here](https://github.com/hrfee/jfa-go/wiki/Build#native). Also, the whole program now compiles into a single file so there is no easy way of modifying the assets (HTML, CSS, etc.) after the program is compiled, but there are some customization features built-in (An email editor, custom branding, etc.).

**Reply by hrfee** [reply]
I see you're using Windows, which the compile process doesn't really support. If you want to, try doing it on a Linux or Mac system.

**Reply by hrfee** [reply]
> Should I go with your other project? named jellyfin-accounts [deprecated]

No, it no longer works. 

> I already know how to run it without mac & linux! cause I installed docker! then I have to turn the dev mode & Linux kernal package! followed by this link via MS - https://docs.microsoft.com/en-us/windows/wsl/install-win10#step-4---download-the-linux-kernel-update-package

If you want to run jfa-go in Docker on your server, then using Docker on windows won't help. You would need to compile the container on the server itself. If you aren't going to be using Docker on the server, then you can compile the software in WSL and then transfer it to the server.

**Reply by hrfee** [reply]
> https://github.com/hrfee/jfa-go/blob/main/go.mod does this just need to do! ???

I don't understand, are you asking what it does? That file stores the list of libraries that the program needs so they can be downloaded when compiling.

**Reply by hrfee** [reply]
> Yeah! I longer going to use docker! I think you made me learn so many new languages! Thank you for making me rude to myself! Really so much patience required!

Don't install Golang on windows, setup WSL first with Debian Linux or something and install it in there.

**Reply by hrfee** [reply]
Sorry, was just writing some instructions and I realised that the version of `go` available on Ubuntu is too old. You'll need to use something like Arch Linux instead.

**Reply by hrfee** [reply]
I've used arch in WSL before using [this](https://wsldl-pg.github.io/ArchW-docs/How-to-Setup/), so once installed you should be able to use it in VSCode.

**Reply by hrfee** [reply]
> I can't run it via VS code? after installing go-in win? Or tell me how I update the `go` in Linux Ubuntu
> Let me see what will happen!
> https://www.tecmint.com/install-go-in-ubuntu/

Ah, that will work too if you change the version from `1.15` to `1.16`.

**Reply by hrfee** [reply]
I use arch exclusively now, so personally I prefer it to Ubuntu, although I don't know what it would be like in WSL,

**Reply by hrfee** [reply]
Ok, I tested this on Ubuntu 20.04:
```shell
sudo apt-get update
sudo apt-get install git python curl wget build-essential
wget https://golang.org/dl/go1.16.linux-amd64.tar.gz
sudo tar -C /usr/local -xvf go1.16.linux-amd64.tar.gz
sudo echo 'export GOPATH=$(go env GOPATH)' >> /etc/profile
sudo echo "export PATH=/usr/local/go/bin:$GOPATH/bin:$PATH" >> /etc/profile
source /etc/profile
curl -fsSL https://deb.nodesource.com/setup_14.x | sudo -E bash -
sudo apt-get install nodejs

git clone https://github.com/hrfee/jfa-go
cd jfa-go
make all
```

The executable will be placed in the `build` directory in the `jfa-go` folder.

**Reply by hrfee** [reply]
I'll look into making one.

**Reply by hrfee** [reply]
Really? The last time I used it the performance was surprisingly good, much better than a normal virtual machine.

**Reply by hrfee** [reply]
Ah, that's weird, can you access the internet within WSL?

**Reply by hrfee** [reply]
> or just add a search button here so whenever someone tells me I forgot my password I will search it by their email & change the password! rather scroll it!

good idea, i guess i never considered a search since i don't have many users.

**Reply by KeyboardDabbler** [reply]
@hrfee 
i am running jfa-go via the executable on ubuntu, which has just worked since install. 
I now would like to try running jfa-go via the docker method. I am running into many problems though, unable to load user, config unauthorized and i can not view the data folder.

How can i view the debug logs when running in docker? i have enabled the debug setting. 

Edit
found the command i was looking for in this [LINK](https://github.com/hrfee/jfa-go/issues/55#issuecomment-778620048)

```
keyboarddabbler@HTPC-01:~$ sudo docker logs jfa-go
jfa-go version: 0.3.2 (8f65e2e)

Socket: /tmp/jfa-go.sock
2021/04/03 19:09:46 Using external storage
[INFO] 19:09:46 Copied default configuration to "/data/config.ini"
[INFO] 19:09:46 Initializing router
[INFO] 19:09:46 Loading routes
[INFO] 19:09:46 Loading setup @ 0.0.0.0:8056
[GIN] GET(/users) => 404
[GIN] GET(/invites) => 404
[GIN] GET(/invites) => 404
[GIN] GET(/users) => 404
[GIN] GET(/) => 200
[GIN] GET(/banner.svg) => 200
[GIN] GET(/css/remixicon.woff2?t=1590207869815) => 200
[GIN] GET(/js/setup.js) => 200
[GIN] GET(/lang/form) => 200
[GIN] GET(/lang/email) => 200
[GIN] GET(/lang/setup) => 200
[GIN] GET(/lang/admin) => 200
[GIN] POST(/jellyfin/test) => 200
[INFO] 19:13:55 Config modification requested
[INFO] 19:13:55 Restarting...
[ERROR] 19:13:55 Failure serving: http: Server closed
[GIN] POST(/config) => 200
jfa-go version: 0.3.2 (8f65e2e)

[INFO] 19:13:55 Using Jellyfin server type
[INFO] 19:13:55 Authenticated with http://192.168.1.208:8096
THIRDERR parsing time "2021-04-02T04:13:01.86332ZZ" as "2006-01-02T15:04:05.0000000+00:00": cannot parse "" as ".0000000"
parsing time ""2021-04-02T04:13:01.86332ZZ"" as ""2006-01-02T15:04:05Z07:00"": cannot parse "Z"" as """
[INFO] 19:13:55 Initializing router
[INFO] 19:13:55 User daemon started
[INFO] 19:13:55 Invite daemon started
[INFO] 19:13:55 Starting password reset daemon
[INFO] 19:13:55 Loading routes
[ERROR] 19:14:05 Failed to get latest tag (0): <nil>
2021/04/03 19:14:05 Failed to authenticate with updater @ https://builds.hrfee.pw: Timed out
[GIN] GET(/) => 200
[GIN] GET(/css/bundle.css) => 304
[GIN] GET(/css/remixicon.css) => 304
[GIN] GET(/js/admin.js) => 200
[GIN] GET(/token/refresh) => 400
[GIN] GET(/lang/admin) => 200
[INFO] 19:14:32 Token requested (login attempt)
[GIN] GET(/token/login) => 200
[GIN] GET(/config/update) => 200
[GIN] GET(/invites) => 200
THIRDERR parsing time "2021-04-02T04:13:01.86332ZZ" as "2006-01-02T15:04:05.0000000+00:00": cannot parse "" as ".0000000"
parsing time ""2021-04-02T04:13:01.86332ZZ"" as ""2006-01-02T15:04:05Z07:00"": cannot parse "Z"" as """
[ERROR] 19:14:39 Failed to get users from Jellyfin (200): parsing time ""2021-04-02T04:13:01.86332ZZ"" as ""2006-01-02T15:04:05Z07:00"": cannot parse "Z"" as """
[GIN] GET(/users) => 500
[INFO] 19:14:41 Config requested
[GIN] GET(/config) => 200
[GIN] GET(/profiles) => 200
THIRDERR parsing time "2021-04-02T04:13:01.86332ZZ" as "2006-01-02T15:04:05.0000000+00:00": cannot parse "" as ".0000000"
parsing time ""2021-04-02T04:13:01.86332ZZ"" as ""2006-01-02T15:04:05Z07:00"": cannot parse "Z"" as """
[ERROR] 19:14:52 Failed to get users from Jellyfin (200): parsing time ""2021-04-02T04:13:01.86332ZZ"" as ""2006-01-02T15:04:05Z07:00"": cannot parse "Z"" as """
[GIN] GET(/users) => 500
```

**Reply by hrfee** [reply]
Sorry i didn't reply sooner, someone else had this issue (I did as well shortly after) and I fixed it in fc6b6a9c6b204c6bb5def8d7a2db8fc0c6960669. You'll have to switch to the unstable builds until I do a new release. You can do that by recreating your docker container with the `hrfee/jfa-go:unstable` image instead.

**Reply by KeyboardDabbler** [reply]
Great work, I can confirm everything is working on the unstable build. I may aswell stay on this build now. I have enabled notifications and will help test new commits.

**Comment by Rapter1099** [comment]
Apologies for the wrong place to post, but I am after some help in regards to email notifications.
The SMTP client I use uses STARTTLS, and (I think) provides a self signed certificate (its self hosted). It looks like jfa does not like this, and therefore refuses to send emails out. Is there a way around this at all please?

My best guess is this needs to be set to true in the email.go file, but I have no idea how to do this and make use of it.
"InsecureSkipVerify: false,"

Any help would be greatly appreciated.

**Reply by hrfee** [reply]
I guess the best (as in most secure) way would be to allow you to add a cert to the trust pool yourself. I'll work on adding the option tonight.

**Reply by hrfee** [reply]
just pushed it, once the build is done you can update and specify the path to a cert in Settings > SMTP > Path to custom SSL cert. I was kinda too lazy to test it so hopefully it should work.

**Reply by hrfee** [reply]
Also added an advanced settings toggle which you'll need to check to see the setting.

**Reply by Rapter1099** [reply]
Nice one thanks! I don't have the chance to test it out at the moment but once I have I will let you know how it went!

**Comment by CyferShepard** [comment]
HI, just coming over from reddit regarding the password resets, needed help with that, not sure if this is the right place. So the issue I get is when someone clicks reset password, it doesn't request a pin or anything, it just resets it to blank. This is where i go to reset.
![image](https://user-images.githubusercontent.com/24864904/118399927-cd73d180-b65f-11eb-9400-e3a425a2059f.png)

**Reply by hrfee** [reply]
Ah i see. Since you're resetting your password while you're logged in, theres no need to send a PIN or anything. The PIN is used when you press "Forgot Password" on the login page. Give that a go and see if a file is created.

**Reply by CyferShepard** [reply]
So theres supposed to be a forgot password button on the login screen? I dont get that option, is it something i need to enable?
![image](https://user-images.githubusercontent.com/24864904/118400141-ab2e8380-b660-11eb-834f-e69ed50f0fa1.png)

**Reply by hrfee** [reply]
It might be your custom CSS, maybe try temporarily disabling it?

**Reply by CyferShepard** [reply]
I thought of that too, so i created a new instance of jellyfin from scratch on my test VM, same thing, no button

**Reply by CyferShepard** [reply]
Okay tried again now, needed to clear cache, see the button now, thanks for the help, ive been going in circles trying to figure this out

**Reply by hrfee** [reply]
You don't use the LDAP plugin, right? Also, are any of your users visible on the login page? If not, try disabling `Hide this user from login screens` on a users profile in the dashboard.

**Reply by CyferShepard** [reply]
No im not using LDAP, I have a small userbase so by default my users are not hidden. I found out its because of the Monochromatic theme im using. it has a css section to hide that button
`#loginPage .raised.cancel.block.btnForgotPassword.emby-button {
  display: none
}`

I just overrode it to display:block and it reappeared.

**Reply by hrfee** [reply]
Ah ok, great. Does the passwordreset file get created, and does it get detected by jfa-go?

**Reply by CyferShepard** [reply]
So the file is created, finally located it, for me it was in `/var/lib/jellyfin/passwordresetxxxxx-xxxxx-xxx.json`.
Just going to reconfigure the path in jfa-go and retest

**Reply by CyferShepard** [reply]
It Works. After so long it finally works. Thanks so much for the guidance.

**Comment by Poncygit** [comment]
I thought i'd give JFA-GO a try. 
 I ran the Windows exe, but the shell window just froze after giving an SMTP crypto error of some description. Also seems no way of deinstalling, as it's not listed in Apps and Features. Possibly because the install failed.

[35mjfa-go version: ←[97m0.3.4←[0m (←[37mb2b5083←[0m)

←[0m

Socket: C:\Users\MEDIAS~1\AppData\Local\Temp\jfa-go.sock

2021/05/16 22:58:04 Using internal storage

[ERROR] 22:58:04 email.go:152: ←[31mError while initiating SMTP mailer: crypto/x509: system root pool is not available on Windows←[0m

[INFO] 22:58:04 ←[97mUsing Jellyfin server type

←[0m
What am I doing wrong?

**Reply by hrfee** [reply]
You didn't do anything wrong. I guess i have less windows users than i thought, this bug has been present for a while and would have broken things for most windows installs. Apparently it's not possible to access the system certificate pool on windows ([here](https://golang.org/src/crypto/x509/cert_pool.go?s=2159:2223) on line 107) so i've disabled the option in 057f306. Should be compiled in ~10 minutes, try downloading it and trying again.

**Reply by Poncygit** [reply]
OK. I'll give it a go again. Cheers

**Comment by Poncygit** [comment]
No. I now get this after eventually finding the file.
Only difference is the Crypto error message has gone.?

←[35mjfa-go version: ←[97mgit←[0m (←[37m18bcd55←[0m)
←[0m
Socket: C:\Users\MEDIAS~1\AppData\Local\Temp\jfa-go.sock
2021/05/17 03:07:16 Using internal storage
[INFO] 03:07:16  ←[97mUsing Jellyfin server type
←[0m

**Reply by hrfee** [reply]
Have you already went through the setup process? It looks like it's hanging on trying to connect to Jellyfin, and if setup wasn't completed then it will try to use the default value, which is a non-existent URL. If you haven't done setup yet, try deleting `C:\Users\MEDIAS~1\AppData\Roaming\jfa-go` and running again.

**Comment by Poncygit** [comment]
I deleted the roaming folder and tried again.
I gave the installer permission for private network then the shell window
stopped...

←[35mjfa-go version: ←[97mgit←[0m (←[37m057f306←[0m)
←[0m
Socket: C:\Users\MEDIAS~1\AppData\Local\Temp\jfa-go.sock
2021/05/17 21:37:56 Using internal storage
[INFO] 21:37:56  ←[97mCopied default configuration to
"C:\Users\MediaServer\AppData\Roaming\jfa-go\config.ini"←[0m
[INFO] 21:37:56  ←[97mInitializing router
←[0m
[INFO] 21:37:56  ←[97mLoading routes
←[0m
[INFO] 21:37:56  ←[97mLoading setup @ 0.0.0.0:8056←[0m

On Mon, 17 May 2021 at 12:46, Harvey Tindall ***@***.***>
wrote:

> Have you already went through the setup process? It looks like it's
> hanging on trying to connect to Jellyfin, and if setup wasn't completed
> then it will try to use the default value, which is a non-existent URL. If
> you haven't done setup yet, try deleting
> C:\Users\MEDIAS~1\AppData\Roaming\jfa-go and running again.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/hrfee/jfa-go/discussions/43#discussioncomment-747632>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AJDUJWK2GJ64GNJPXIR7AX3TOD6YJANCNFSM4XFSV2RA>
> .
>


-- 
http://www.destinationmiltonkeynes.co.uk/
https://www.bletchleypark.org.uk/
http://www.miltonkeynesmuseum.org.uk/
http://bentleypriorymuseum.org.uk/

**Reply by hrfee** [reply]
That's intended, visit `localhost:8056` in your browser to start the setup.

**Comment by Poncygit** [comment]
Ah OK thanks.  Maybe a confidence notice to say completed?
Now to try...
Cheers, Mick. 👍👍

On Mon, 17 May 2021 at 21:50, Harvey Tindall ***@***.***>
wrote:

> That's intended, visit localhost:8056 in your browser to start the setup.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/hrfee/jfa-go/discussions/43#discussioncomment-749982>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AJDUJWJGU45W6X5P5SGFH33TOF6PTANCNFSM4XFSV2RA>
> .
>


-- 
http://www.destinationmiltonkeynes.co.uk/
https://www.bletchleypark.org.uk/
http://www.miltonkeynesmuseum.org.uk/
http://bentleypriorymuseum.org.uk/

**Reply by hrfee** [reply]
Will do, its confused a decent amount of users.

**Comment by Poncygit** [comment]
Nope, nothing with browser.
I checked running processes, JF is running, but no sign of JFA

On Mon, 17 May 2021 at 22:00, Harvey Tindall ***@***.***>
wrote:

> Will do, its confused a decent amount of users.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/hrfee/jfa-go/discussions/43#discussioncomment-750030>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AJDUJWI52ST7G43RJPXZEWTTOF7YTANCNFSM4XFSV2RA>
> .
>


-- 
http://www.destinationmiltonkeynes.co.uk/
https://www.bletchleypark.org.uk/
http://www.miltonkeynesmuseum.org.uk/
http://bentleypriorymuseum.org.uk/

**Reply by hrfee** [reply]
You need to keep the command line open, as jfa-go doesn't run in the background by default. You'll also need to delete that AppData folder again since setup wasn't completed.

If you don't mind switching to a nightly build, yesterday I added a tray app that that will run in the background and allow you to easily start & stop it. There's also an option to start up automatically on login. You can download it [here](https://builds.hrfee.pw/view/hrfee/jfa-go) (get the file with `TrayIcon` and `Windows` in the name), and then do either of these:
* open a terminal and run the executable with the `tray` argument.
* create a shortcut to the executable, right click > properties, and add ` tray` to the file path. This way you can just click the shortcut to start.

If you want to stay on a stable version, or want it to start at boot instead of login, I've heard something called NSSM is quite common for running things in the background, although I can't provide much help with it.

**Reply by hrfee** [reply]
Oh sorry, didn't see the part where you said "shell window stopped", that sounds like a problem. Could you open the folder containing the jfa-go executable in the file explorer, hold shift and rick-click, then press `Open Powershell window here`. A terminal will open, type `jfa-go`, then press Tab to autocomplete, then press Enter. This way the terminal should stay open it jfa-go crashses and you'll be able to see the error.

**Comment by Poncygit** [comment]
Thanks. I did see the tray app in the nightly build. I'll try that one.
TVM


On Mon, 17 May 2021 at 22:12, Harvey Tindall ***@***.***>
wrote:

> You need to keep the command line open, as jfa-go doesn't run in the
> background by default. You'll also need to delete that AppData folder again
> since setup wasn't completed.
>
> If you don't mind switching to a nightly build, yesterday I added a tray
> app that that will run in the background and allow you to easily start &
> stop it. There's also an option to start up automatically on login. You can
> download it here <https://builds.hrfee.pw/view/hrfee/jfa-go> (get the
> file with TrayIcon and Windows in the name), and then do either of these:
>
>    - open a terminal and run the executable with the tray argument.
>    - create a shortcut to the executable, right click > properties, and
>    add tray to the file path. This way you can just click the shortcut to
>    start.
>
> If you want to stay on a stable version, or want it to start at boot
> instead of login, I've heard something called NSSM is quite common for
> running things in the background, although I can't provide much help with
> it.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/hrfee/jfa-go/discussions/43#discussioncomment-750067>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AJDUJWOMROCUCJUP4AXSR7DTOGBFTANCNFSM4XFSV2RA>
> .
>


-- 
http://www.destinationmiltonkeynes.co.uk/
https://www.bletchleypark.org.uk/
http://www.miltonkeynesmuseum.org.uk/
http://bentleypriorymuseum.org.uk/

**Comment by Poncygit** [comment]
At one time, I had shift PS, but not anymore. I tried the time honoured
registry hack, but it still doesn't show up even with a restart.
Anyway, I left the window open and have accessed via browser page. Going
through the setup, but not having success with the test to JF  with new
admin account.
I'll keep banging away and see how far I get. No tray icon seen yet.

**Reply by hrfee** [reply]
If the shift right click doesn't work, you could try the other method or just copy the path to the folder from the file explorer, then open a shell yourself and run `cd <path>`.

**Comment by Poncygit** [comment]
OK, looking good. Not too sure about the tray icon, I need to do a bit more
work on that, otherwise its working with test emails.
I'll play with it later on, time for a few Z'dz.😴😴
Thanks for your work. Much appreciated.
Cheers,


On Mon, 17 May 2021 at 23:16, Harvey Tindall ***@***.***>
wrote:

> If the shift right click doesn't work, you could try the other method or
> just copy the path to the folder from the file explorer, then open a shell
> yourself and run cd <path>.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/hrfee/jfa-go/discussions/43#discussioncomment-750232>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AJDUJWMS2D32FCEEA4VJRXDTOGISHANCNFSM4XFSV2RA>
> .
>


-- 
http://www.destinationmiltonkeynes.co.uk/
https://www.bletchleypark.org.uk/
http://www.miltonkeynesmuseum.org.uk/
http://bentleypriorymuseum.org.uk/

**Reply by Poncygit** [reply]
OK. All appear to work for me that I have tried.
I've got the tray icon running, but what do I do with the shell dialogue?
I've set to minimise on launch, but obviously that's not good.
As a matter of interest, is there anyway this could be made a JF Plugin? 
Otherwise well done. There's a few things here and there that might need tweeking to make it less grey and more black and white on what's being setup or enabled as I see it. To anyone else, it's probably plain as day!
😴😴

**Reply by hrfee** [reply]
Does the shell appear if you launch it through a shortcut? The executable is compiled to not automatically open a shell window when executed through the GUI so hopefully it shouldn't.

**Comment by Poncygit** [comment]
Shell appears when launched directly. No Tray Icon
Shell and Tray Icon appears when launched with a short cut...
C:\JFA-GO\jfa-go.exe tray
Tray Icon ticked to run on login.
Log in to what? Cannot JAF-GO be set to Autostart same as JF Tray Icon ?
confused.com on your intention 😀



On Fri, 21 May 2021 at 18:30, Harvey Tindall ***@***.***>
wrote:

> Does the shell appear if you launch it through a shortcut? The executable
> is compiled to not automatically open a shell window when executed through
> the GUI so hopefully it shouldn't.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/hrfee/jfa-go/discussions/43#discussioncomment-769421>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AJDUJWLVOVVRF2DFIQFAUKTTO2KETANCNFSM4XFSV2RA>
> .
>


-- 
http://www.destinationmiltonkeynes.co.uk/
https://www.bletchleypark.org.uk/
http://www.miltonkeynesmuseum.org.uk/
http://bentleypriorymuseum.org.uk/

**Reply by hrfee** [reply]
It means run when you log in to your computer.

And my mistake for the terminal appearing, I only disabled it for the build system I use in development, not the one used for releasing binaries. I'll change it now.

**Comment by Poncygit** [comment]
I would like the ability to send my own text with messages such as the welcome message, instead of the default.

**Reply by hrfee** [reply]
Press the little pen icon next to the "Emails" (or "Messages" if you're on the nightly), you can edit them with markdown.

**Comment by MauricioDoigo** [comment]
Hello!
I apologize if what I'm posting is repeated. I did a search and didn't find anything like that.
After installation, it ran perfectly, but did not start the system.
Thinking about helping anyone who has the same problem, create a bash to install, and configure automatic start on the system.

Note: I'm using it on a raspberry pi3 with raspios.

```
bash -c '
set -e

# Solicita o nome do usuário
read -p "Digite o nome do usuário para configurar o jfa-go: " USERNAME

# Pergunta se deseja a versão estável ou instável
echo "Escolha a versão do jfa-go:"
echo "1) Estável"
echo "2) Instável"
read -p "Digite 1 ou 2: " VERSION_CHOICE

# Define o repositório com base na escolha
if [ "$VERSION_CHOICE" -eq 1 ]; then
    REPO="deb https://apt.hrfee.dev trusty main"
elif [ "$VERSION_CHOICE" -eq 2 ]; then
    REPO="deb https://apt.hrfee.dev trusty-unstable main"
else
    echo "Escolha inválida. Usando a versão estável por padrão."
    REPO="deb https://apt.hrfee.dev trusty main"
fi

# Atualiza e instala dependências
echo "Atualizando pacotes e instalando dependências..."
sudo apt-get update
sudo apt-get install -y curl apt-transport-https gnupg

# Adiciona a chave do repositório
echo "Adicionando chave do repositório..."
curl https://apt.hrfee.dev/hrfee.pubkey.gpg | gpg --dearmor | sudo tee /etc/apt/trusted.gpg.d/apt.hrfee.dev.gpg

# Adiciona o repositório
echo "Adicionando repositório do jfa-go..."
echo "$REPO" | sudo tee /etc/apt/sources.list.d/hrfee.list

# Atualiza a lista de pacotes
echo "Atualizando lista de pacotes..."
sudo apt-get update

# Instala o jfa-go
echo "Instalando jfa-go..."
sudo apt-get install -y jfa-go

# Cria e configura o serviço systemd
echo "Configurando serviço systemd para jfa-go..."
sudo tee /etc/systemd/system/jfa-go.service > /dev/null <<EOL
[Unit]
Description=jfa-go Service
After=network.target

[Service]
ExecStart=/usr/bin/jfa-go -c /home/$USERNAME/.config/jfa-go/config.ini
Restart=on-failure
User=$USERNAME
Group=$USERNAME

[Install]
WantedBy=multi-user.target
EOL

# Configura permissões
echo "Configurando permissões..."
sudo chown $USERNAME:$USERNAME /usr/bin/jfa-go
sudo chmod +x /usr/bin/jfa-go
sudo mkdir -p /home/$USERNAME/.config/jfa-go
sudo chown -R $USERNAME:$USERNAME /home/$USERNAME/.config/jfa-go
sudo chmod -R 755 /home/$USERNAME/.config/jfa-go

# Recarrega o daemon systemd e inicia o serviço
echo "Recarregando daemon systemd e iniciando o serviço..."
sudo systemctl daemon-reload
sudo systemctl enable jfa-go
sudo systemctl start jfa-go

# Verifica o status do serviço
echo "Verificando o status do serviço jfa-go..."
sudo systemctl status jfa-go

echo "Configuração concluída com sucesso!"
'

```

**Comment by baervers23** [comment]
**feature: implement custom API requests** (for example sync accounts to other apps)

your project is to complicated for me, but i would like to have this feature so ill try to help ^^ got an example on how to custom api... 

[feature-customapi.zip](https://github.com/user-attachments/files/21415460/feature-customapi.zip)

sorry im still learning, hope you like the idea

**Comment by Pedro-748** [comment]
Feature?: Is it possible to require EITHER email or discord, instead of requiring one or the other/ or both. I don't care which they use as long as they set up at least one.
